src/libraries/System.Net.Http/src/System/Net/Http/Headers/CacheControlHeaderParser.cs

```diff
@@ -22,7 +23,22 @@ private CacheControlHeaderParser()
         protected override int GetParsedValueLength(string value, int startIndex, object? storeValue,
             out object? parsedValue)
         {
-            CacheControlHeaderValue? temp = storeValue as CacheControlHeaderValue;
+            CacheControlHeaderValue? temp = null;
+            if (storeValue is List<object> list)
+            {
+                foreach (object item in list)
+                {
+                    if (item is CacheControlHeaderValue cacheControl)
+                    {
+                        temp = cacheControl;
+                        break;
+                    }
+                }
+            }
+            else
+            {
+                temp = storeValue as CacheControlHeaderValue;
+            }
             Debug.Assert(storeValue == null || temp != null, "'storeValue' is not of type CacheControlHeaderValue");
```

Comment:
This assert isn't valid anymore. `storeValue` could be `InvalidValue` or `List<object>` even if `temp is null`.